cpp/models/abm/common_abm_loggers.h

```diff
@@ -0,0 +1,182 @@
+    }
+};
+
+mio::abm::ActivityType guess_activity_type(mio::abm::LocationType current_location)
+{
+    switch (current_location) {
+    case mio::abm::LocationType::Home:
+        return mio::abm::ActivityType::Home;
+    case mio::abm::LocationType::Work:
+        return mio::abm::ActivityType::Workplace;
+    case mio::abm::LocationType::School:
+        return mio::abm::ActivityType::Education;
+    case mio::abm::LocationType::SocialEvent:
+        return mio::abm::ActivityType::Leisure;
+    case mio::abm::LocationType::BasicsShop:
+        return mio::abm::ActivityType::Shopping;
+    case mio::abm::LocationType::ICU:
+        return mio::abm::ActivityType::OtherActivity;
+    case mio::abm::LocationType::Hospital:
+        return mio::abm::ActivityType::OtherActivity;
+    case mio::abm::LocationType::Cemetery:
+        return mio::abm::ActivityType::OtherActivity;
+    default:
+        return mio::abm::ActivityType::UnknownActivity;
+    }
+}
+
+struct LogDataForMovement : mio::LogAlways {
+    using Type = std::vector<std::tuple<uint32_t, uint32_t, mio::abm::TimePoint, mio::abm::TransportMode,
+                                        mio::abm::ActivityType, mio::abm::InfectionState>>;
+    static Type log(const mio::abm::Simulation& sim)
+    {
+        Type movement_data{};
+        for (Person p : sim.get_world().get_persons()) {
+            movement_data.push_back(std::make_tuple(
+                p.get_person_id(), p.get_location().get_index(), sim.get_time(), p.get_last_transport_mode(),
+                guess_activity_type(p.get_location().get_type()), p.get_infection_state(sim.get_time())));
+        }
+        return movement_data;
+    }
+};
+
+/*
+* @brief This is like the DataWriterToMemory, but it only logs time series data.
+* @tparam Loggers The loggers that are used to log data. The loggers must return a touple with a TimePoint and a value.
+*/
+template <class... Loggers>
+struct TimeSeriesWriter {
+    using Data = std::tuple<mio::TimeSeries<ScalarType>>;
+    template <class Logger>
+    static void log_this(const typename Logger::Type& t, Data& data)
+    {
+        std::get<details::index_templ_pack<Logger, Loggers...>()>(data).add_time_point(t.first.days(), t.second);
+    }
+};
+
+/*
+* @brief Logger to log the TimeSeries of the number of Person%s in an #InfectionState.
+*/
+struct LogInfectionState : mio::LogAlways {
+    using Type = std::pair<mio::abm::TimePoint, Eigen::VectorXd>;
+    static Type log(const mio::abm::Simulation& sim)
+    {
+
+        Eigen::VectorXd sum = Eigen::VectorXd::Zero(Eigen::Index(mio::abm::InfectionState::Count));
+        PRAGMA_OMP(for)
+        for (auto i = size_t(0); i < sim.get_world().get_locations().size(); ++i) {
+            auto&& location = sim.get_world().get_locations()[i];
+            sum += location.get_subpopulations().get_last_value().cast<ScalarType>();
+        }
+        return std::make_pair(sim.get_time(), sum);
+    }
+};
```

Comment:
Since we have this now, we should delete the `TimeSeries m_subpopulations` in the locations and all related functions, too, and directly implement here what is in `Location::store_subpopulations` (or do this in a function `Location::get_current_subpopulations`, which would probably be better). This entry is very large and was only there to save the results for the final entry in the simulation subpopulation TimeSeries, which you now correctly deleted.

Reply:
FYI there exists a function size_t Location::get_subpopulation(TimePoint t, InfectionState state) const which probably does what you mean. I kept that.